### ISSUE TITLE
Why are properties with null values removed?

### ISSUE THREAD

**Issue by bravius** [bitnine-oss/agensgraph#271]
Just wondering what the AgensGraph rationale is for not allowing properties with null values?  

This behavior seems odd because null is a valid JSON primitive, and therefore should be valid in any JSON document stored in AgensGraph.

**Comment by gtyun** [comment]
@bravius Hi bravius, sorry for late response.

In past, null values was allowed. But there was some problems there.
1. According to Cypher grammar, SET null means remove keys but AgensGraph's CREATE null make the key with null values. It is not consistent.
2. The nonexistent keys and null valued keys casue confusion like following..

```
a { name : 'A'}
b { name : 'B', key:null }

MATCH (n {key:null}) RETURN n;
-> b returned
MATCH (n) WHERE key IS NULL RETURN n;
-> a, b, returned.
```
This is why we decided to prohibit null valued keys. 

But recently, we noticed that null values can be important in ARRAY. (Actually Agens does not support array in past). So the presence of a null value key will be discussed.
